ros/angel_system_nodes/angel_system_nodes/hl2ss_ros_bridge.py

```diff
@@ -0,0 +1,316 @@
+    HandJointPose,
+    HandJointPosesUpdate,
+)
+from angel_system.hl2ss.viewer import hl2ss
+from angel_utils.conversion import hl2ss_stamp_to_ros_time
+
+
+BRIDGE = CvBridge()
+
+
+class HL2SSROSBridge(Node):
+    """
+    ROS node that uses HL2SS client/server library to convert HL2SS data to
+    ROS messages used throughout the ANGEL system.
+    """
+
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+
+        # Declare ROS topics
+        self._image_topic = (
+            self.declare_parameter("image_topic", "")
+            .get_parameter_value()
+            .string_value
+        )
+        self._hand_pose_topic = (
+            self.declare_parameter("hand_pose_topic", "")
+            .get_parameter_value()
+            .string_value
+        )
+        self.ip_addr = (
+            self.declare_parameter("ip_addr", "")
+            .get_parameter_value()
+            .string_value
+        )
+
+        log = self.get_logger()
+        if self._image_topic == "":
+            raise ValueError("Please provide the image topic with the `image_topic` parameter")
+        if self._hand_pose_topic == "":
+            raise ValueError("Please provide the hand pose topic with the `hand_pose_topic` parameter")
+        if self.ip_addr == "":
+            raise ValueError("Please provide HL2 IPv4 address with the `ip_addr` parameter")
+
+
+        log.info(f"Image topic: {self._image_topic}")
+        log.info(f"Hand pose topic: {self._hand_pose_topic}")
+        log.info(f"HL2 IP: {self.ip_addr}")
+
+        # Define HL2SS server ports
+        self.pv_port = hl2ss.StreamPort.PERSONAL_VIDEO
+        self.si_port = hl2ss.StreamPort.SPATIAL_INPUT
+
+        # Create frame publisher
+        self.frame_publisher = self.create_publisher(
+            Image,
+            self._image_topic,
+            1
+        )
+        # Create the hand joint pose publisher
+        self.hand_publisher = self.create_publisher(
+            HandJointPosesUpdate,
+            self._hand_pose_topic,
+            1
+        )
+
+        log.info("Connecting to HL2SS servers...")
+        self.connect_hl2ss_pv()
+        log.info("PV client connected!")
+        self.connect_hl2ss_si()
+        log.info("SI client connected!")
+
+        log.info("Starting publishing threads...")
+        # Start the frame publishing thread
+        self._pv_active = Event()
+        self._pv_active.set()
+        self._pv_thread = Thread(
+            target=self.pv_publisher,
+            name="publish_frames"
+        )
+        self._pv_thread.daemon = True
+        self._pv_thread.start()
+        # Start the hand tracking data thread
+        self._si_active = Event()
+        self._si_active.set()
+        self._si_thread = Thread(
+            target=self.si_publisher,
+            name="publish_frames"
+        )
+        self._si_thread.daemon = True
+        self._si_thread.start()
+        log.info("Starting publishing threads... Done")
+
```

Comment:
extraneous newlines between methods. I/we need to add flake8 or black to CI this stuff... Generally there should be a single newline between intra-class methods.